### PR TITLE
[MM-29564] Add end to end tests to the "moveCommandCmdF" 

### PR DESCRIPTION
**Pull request by needkopi**
#### Summary

add e2e testing for `moveCommandCmdF`

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/15942
Jira https://mattermost.atlassian.net/browse/MM-29564

### CODE REVIEW

File: commands/command_e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	s.RunForAllClients("move command to non existing team", func(c client.Client) {
+		printer.Clean()
+
+		team := "nonexisting team"
+		err := moveCommandCmdF(s.th.SystemAdminClient,
+			&cobra.Command{},
+			[]string{team, command.Id})
+		s.Require().NotNil(err)
+		s.Require().Equal(fmt.Sprintf("unable to find team '%s'", team), err.Error())
+	})
```
Comment (isacikgoz) [inline commands/command_e2e_test.go:39]: This is not using the input function's `client.Client`. The case is always executed with `s.th.SystemAdminClient`. This needs to be fixed and this applies to other cases/functions as well.

File: commands/command_e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+	s.RunForAllClients("move non existing command", func(c client.Client) {
```
Comment (isacikgoz) [inline commands/command_e2e_test.go:41]: Did you wanted to mean `channel` here?

File: commands/command_e2e_test.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+		printer.Clean()
+
+		err := moveCommandCmdF(s.th.SystemAdminClient,
+			&cobra.Command{},
+			[]string{s.th.BasicTeam.Name, "nothing"})
+		s.Require().NotNil(err)
+		s.Require().Equal("unable to find command 'nothing'", err.Error())
+	})
+
+	s.RunForAllClients("move existing command to existing team", func(c client.Client) {
```
Comment (isacikgoz) [inline commands/command_e2e_test.go:51]: I think we should add a separate test case for regular users and `SystemAdmin` and `Local` clients. We need to be sure if our end to end tests leak any permission related issues.

**Review by isacikgoz — CHANGES_REQUESTED** [review]
Thanks for the contribution @needkopi 👍 I've requested some changes. Let me know if you need additional explanation.

File: commands/command_e2e_test.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+			&cobra.Command{},
+			[]string{team, command.Id})
+		s.Require().NotNil(err)
+		s.Require().Equal(fmt.Sprintf("unable to find team '%s'", team), err.Error())
+	})
+
+	s.RunForAllClients("move non existing command", func(c client.Client) {
+		printer.Clean()
+
+		err := moveCommandCmdF(s.th.SystemAdminClient,
```
Comment (mgdelacroix) [inline commands/command_e2e_test.go:44]: Same as in the previous test case and comment from @isacikgoz, this should be:

```suggestion
		err := moveCommandCmdF(c,
```

**Review by mgdelacroix — CHANGES_REQUESTED** [review]

**Comment by jasonblais** [comment]
@needkopi thanks for the contribution! Let us know if you have any questions about the above feedback? We'd be happy to help :)

**Comment by isacikgoz** [comment]
Hey @needkopi! Let us know if you need anything, we'd be happy to help :)

File: commands/command_e2e_test.go (hunk 1 above)
Comment (needkopi) [reply commands/command_e2e_test.go:39]: ah i see, oke @isacikgoz I'll fix it

File: commands/command_e2e_test.go (hunk 2 above)
Comment (needkopi) [reply commands/command_e2e_test.go:41]: I still don't understand what the `channel` means in your reply

File: commands/command_e2e_test.go (hunk 3 above)
Comment (needkopi) [reply commands/command_e2e_test.go:51]: okay I'll add testing for this case

**Comment by needkopi** [comment]
i am sorry @isacikgoz @jasonblais  , yesterday I had a work deadline in my office. now that it's finished maybe this is the right time for me to continue.

**Comment by jfrerich** [comment]
Hi @needkopi!  Thanks for the PR!  I just wanted to check in to see if you have any updates

**Comment by emilyacook** [comment]
Hi @needkopi! Could you let me know what your availability is on this so I can update it per our [inactive contribution guidelines](https://developers.mattermost.com/contribute/getting-started/inactive-contributions/)?

**Comment by lindy65** [comment]
Hi @needkopi - I'm helping to review stale PRs and, since we haven't heard from you in a few months, I'll close this PR for now. Please feel free to re-open if you're able to prioritize. Thanks!

I've re-added the "Up for Grabs" label on https://github.com/mattermost/mattermost-server/issues/15942

//cc @mgdelacroix @isacikgoz

**Comment by rdanielwetan** [comment]
can this issue be assigned to me please? thanks

**Pull request closed without merging**